tasty-query/shared/src/main/scala/tastyquery/Types.scala

```diff
@@ -844,14 +859,17 @@ object Types {
       new PackageRef(packageSym)
   }
 
-  case class TypeRef(override val prefix: Type, private var myDesignator: Designator) extends NamedType {
+  final class TypeRef private (val prefix: Type, myDesignator: TypeName | TypeSymbol | Scala2ExternalSymRef)
```

Comment:
now `myDesignator` is immutable, I think it should still be nulled out or replaced by the type symbol when calculated - otherwise we retain the `Scala2ExternalSymRef` (or `LookupIn` for TermRef)

Reply:
then the `designatorInternal` makes sense to have a scary name, now its just returning a stable field

Reply:
OK, I reintroduced storing the symbol in the designator. This will also indirectly help with `toString`-based debugging.